src/generators/ast.ts

```diff
@@ -245,28 +338,37 @@ function parseTypeSpecificFields(raw: JSONSchema7, type: Type): TypeSpecificFiel
   } else if (type === Type.ARRAY) {
     const fields: ArrayTypeFields = { type, items: { type: Type.ANY } };
     if (typeof raw.items !== 'boolean' && raw.items !== undefined) {
-      // `items` can be a single schemas, or an array of schemas, so standardize on an array.
-      const definitions = raw.items instanceof Array ? raw.items : [raw.items];
-
-      // Convert from JSONSchema7Definition -> JSONSchema7
-      const schemas = definitions.filter((def) => typeof def !== 'boolean') as JSONSchema7[];
-
-      if (schemas.length === 1) {
-        const schema = schemas[0];
-
-        if (
-          schema.properties &&
-          Object.keys(schema.properties).length > 0 &&
-          schema.type === undefined
-        ) {
-          schema.type = ['array', 'object'];
+      // Check if items has a direct $ref
+      if ('$ref' in raw.items && typeof raw.items.$ref === 'string') {
```

Comment:
Shorten this check as well ?
```suggestion
      if (typeof raw.items?.$ref === 'string') {
```